Flaky test: `BraveServiceIntegrationTest.childCompletesBeforeServer`

https://ci.appveyor.com/project/line/armeria/builds/33726346/job/a0rrj3t4qfva6v6f#L1682
```
BraveServiceIntegrationTest > childCompletesBeforeServer FAILED
    java.lang.AssertionError: Expected {"traceId":"323fe8defcbd845a","parentId":"323fe8defcbd845a","id":"b216d181a49390e8","name":"child","timestamp":1593058969521364,"duration":11,"localEndpoint":{"serviceName":"BraveServiceIntegrationTest","ipv4":"127.0.0.1"}} to finish after 1593058969521333
        at brave.test.ITRemote.assertSpanInInterval(ITRemote.java:185)
        at brave.test.http.ITHttpServer.childCompletesBeforeServer(ITHttpServer.java:210)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
        at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
        at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:288)
        at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:282)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.lang.Thread.run(Thread.java:748)
```

--- a/brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java
+++ b/brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java
@@ -20,6 +20,8 @@ import static com.linecorp.armeria.common.HttpStatus.BAD_REQUEST;
 import static com.linecorp.armeria.common.HttpStatus.OK;
 import static org.assertj.core.api.Assertions.assertThat;
 
+import java.io.IOException;
+import java.time.Duration;
 import java.util.concurrent.CompletableFuture;
 import java.util.function.Consumer;
 
@@ -84,8 +86,16 @@ public class BraveServiceIntegrationTest extends ITHttpServer {
         // "/nested" left out as there's no sub-routing feature at the moment.
 
         sb.service("/child", (ctx, req) -> {
+
+            // The `Span.start()` and `Span.finish()` estimate the current time using
+            // `baseEpochMicros + (System.nanoTime() - baseTickNanos)`.
+            // However, Armeria achieves the current time using `System.currentTimeMillis() * 1000` in Java8.
+            // The two different way of getting the current time could cause an error of a millisecond.
+            // Sometimes the tolerance makes the server appear to finish earlier than child in test case.
+            // This is not a problem in the Brave{Service,Client} because they only use Armeria timestamp.
             tracing.tracer().nextSpan().name("child").start().finish();
-            return HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "happy");
+            return HttpResponse.delayed(
+                    HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "happy"), Duration.ofMillis(1));
         });
         sb.service("/baggage", (ctx, req) -> {
             final String value = String.valueOf(BAGGAGE_FIELD.getValue());

--- a/brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java
+++ b/brave/src/test/java/com/linecorp/armeria/server/brave/BraveServiceIntegrationTest.java
@@ -20,7 +20,6 @@ import static com.linecorp.armeria.common.HttpStatus.BAD_REQUEST;
 import static com.linecorp.armeria.common.HttpStatus.OK;
 import static org.assertj.core.api.Assertions.assertThat;
 
-import java.io.IOException;
 import java.time.Duration;
 import java.util.concurrent.CompletableFuture;
 import java.util.function.Consumer;
@@ -38,6 +37,7 @@ import com.linecorp.armeria.common.HttpStatus;
 import com.linecorp.armeria.common.MediaType;
 import com.linecorp.armeria.common.SessionProtocol;
 import com.linecorp.armeria.common.brave.RequestContextCurrentTraceContext;
+import com.linecorp.armeria.common.util.SystemInfo;
 import com.linecorp.armeria.server.Server;
 import com.linecorp.armeria.server.ServerBuilder;
 
@@ -91,11 +91,16 @@ public class BraveServiceIntegrationTest extends ITHttpServer {
             // `baseEpochMicros + (System.nanoTime() - baseTickNanos)`.
             // However, Armeria achieves the current time using `System.currentTimeMillis() * 1000` in Java8.
             // The two different way of getting the current time could cause an error of a millisecond.
-            // Sometimes the tolerance makes the server appear to finish earlier than child in test case.
-            // This is not a problem in the Brave{Service,Client} because they only use Armeria timestamp.
+            // Sometimes the tolerance makes the server appear to finish earlier than child in this test case.
+            // This is not a problem in the Brave{Service,Client} because they only use Armeria's timestamp.
             tracing.tracer().nextSpan().name("child").start().finish();
-            return HttpResponse.delayed(
-                    HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "happy"), Duration.ofMillis(1));
+            final Duration delay;
+            if (SystemInfo.javaVersion() > 8) {
+                delay = Duration.ofNanos(1000);
+            } else {
+                delay = Duration.ofMillis(1);
+            }
+            return HttpResponse.delayed(HttpResponse.of(OK, MediaType.PLAIN_TEXT_UTF_8, "happy"), delay);
         });
         sb.service("/baggage", (ctx, req) -> {
             final String value = String.valueOf(BAGGAGE_FIELD.getValue());

